tests/api/cross_access/test_view.py

```diff
@@ -370,3 +383,121 @@ def test_create_requests_fail_for_over_60_day_start_date(self):
+        """Test updating an entire CAR."""
+        self.data4create["start_date"] = self.format_date(self.ref_time + timedelta(3))
+        self.data4create["end_date"] = self.format_date(self.ref_time + timedelta(5))
+        self.data4create["roles"] = ["role_8", "role_9"]
+
+        car_uuid = self.request_1.request_id
+        url = reverse("cross-detail", kwargs={"pk": str(car_uuid)})
+
+        client = APIClient()
+        response = client.put(
+            url,
+            self.data4create,
+            format="json",
+            **self.associate_admin_request.META
+        )
+
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        for field in self.data4create:
+            if field == "roles":
+                for role in response.data.get("roles"):
+                    self.assertIn(role.get("display_name"), self.data4create["roles"])
+                continue
+            self.assertEqual(self.data4create.get(field), response.data.get(field))
+
+    def test_update_request_fail_acct(self):
+        """Test that updating the account of a CAR fails."""
+        self.data4create["target_account"] = "10001"
+
+        car_uuid = self.request_1.request_id
+        url = reverse("cross-detail", kwargs={"pk": car_uuid})
+
+        client = APIClient()
+        response = client.put(
+            url,
+            self.data4create,
+            format="json",
+            **self.associate_admin_request.META
+        )
+        
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+
+    def test_update_request_expired(self):
+        """Test that updating an expired CAR creates a new one."""
+        self.data4create["end_date"] = self.format_date(self.ref_time + timedelta(20))
+
+        car_uuid = self.request_5.request_id
+        url = reverse("cross-detail", kwargs={"pk": car_uuid})
+
+        client = APIClient()
+        response = client.put(
+            url,
+            self.data4create,
+            format="json",
+            **self.associate_admin_request.META
+        )
+        
+        self.assertEqual(response.status_code, status.HTTP_201_CREATED)
+        self.assertNotEqual(response.data.get("request_id"), car_uuid)
+    
+    def test_partial_update_success(self):
+        """Test updating part of a CAR."""
+        update_data = {
+            "start_date": self.format_date(self.ref_time + timedelta(2))
+        }
+
+        car_uuid = self.request_1.request_id
+        url = reverse("cross-detail", kwargs={"pk": str(car_uuid)})
+
+        client = APIClient()
+        response = client.patch(
+            url,
+            update_data,
+            format="json",
+            **self.associate_admin_request.META
+        )
+
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(response.data.get("start_date"), update_data.get("start_date"))
+
+    def test_partial_update_fail_invalid_field(self):
```

Comment:
Pretty reasonable unit test coverage here. I think I might like to see a little coverage with bad request body or invalid data for both PUT and PATCH. For example:
* Body with extra fields
* Body with fields whose types do not match the spec
* In cases where we take URL params, invalid or mangled parameter values

Historically we've done this type of testing in our integration test suite, but I think they make more sense as unit tests. Also, the generated clients don't easily lend themselves to testing with non-spec inputs, so we may have more flexibility in the unit test coding than we would with the limitations of the IQE framework. If it makes sense we could do these unit tests in a follow-up PR. Alternatively, I we could create a PR with just unit tests that targets this branch.

Reply:
Can do! I'll slip those in tomorrow